system/Validation/FormatRules.php

```diff
@@ -346,4 +346,33 @@ public function valid_date(?string $str = null, ?string $format = null): bool
 
         return $date !== false && $errors !== false && $errors['warning_count'] === 0 && $errors['error_count'] === 0;
     }
+
+    /**
+     * Token JWT Validator
+     *
+     * @param string $str String
+     */
+    public function valid_jwt(?string $str = null): bool
+    {
+        if (empty($str)) {
+            return false;
+        }
+
+        $segments = explode('.', $str);
+
+        if (count($segments) !== 3 || in_array('', $segments, true)) {
+            return false;
+        }
+
+        foreach ($segments as $index => $value) {
+            if ($index < 2) {
+                $encoded = rtrim(base64_encode(base64_decode($value, true)), '==');
+                if ($encoded !== $value) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
```

Comment:
Is there anything else we can check about the third segment? Minimum signature length, or a regex of characters?

Reply:
In algo SHA256-SHA512 & Sodium, get min `32` and max `64` byte.

Ref : [firebase/php-jwt](https://github.com/firebase/php-jwt/blob/main/src/JWT.php#L55) & [JWT.io](https://jwt.io/libraries?language=PHP)